plugins/api-docs/src/components/GraphQlDefinitionWidget/GraphQlDefinition.tsx

```diff
@@ -46,17 +53,46 @@ type Props = {
 
 export const GraphQlDefinition = ({ definition }: Props) => {
   const classes = useStyles();
+  const { entity } = useEntity();
   const schema = buildSchema(definition);
 
+  const url = entity.metadata.annotations?.['backstage.io/api-graphql-url'];
+
   return (
     <div className={classes.root}>
       <div className={classes.graphiQlWrapper}>
         <GraphiQL
-          fetcher={() => Promise.resolve(null) as any}
+          fetcher={
+            url
+              ? async (params: any, options: any = {}) => {
+                  const body = JSON.stringify(params);
+                  const headers = {
+                    'Content-Type': 'application/json',
+                    ...options.headers,
+                  };
+
+                  const res = await fetch(url!, {
```

Comment:
frontend components should prefer to use the fetch that comes out of `useApi(fetchApiRef)` instead of the native one